plugins/woocommerce-admin/client/products/fields/variations/variations.tsx

```diff
@@ -29,9 +32,16 @@ import './variations.scss';
  */
 const DEFAULT_PER_PAGE_OPTION = 25;
 
+const NOT_VISIBLE_TEXT = __( 'Not visible to customers', 'woocommerce' );
+const VISIBLE_TEXT = __( 'Visible to customers', 'woocommerce' );
+const UPDATING_TEXT = __( 'Updating product variation', 'woocommerce' );
+
 export const Variations: React.FC = () => {
 	const [ currentPage, setCurrentPage ] = useState( 1 );
 	const [ perPage, setPerPage ] = useState( DEFAULT_PER_PAGE_OPTION );
+	const [ isUpdating, setIsUpdating ] = useState< Record< string, boolean > >(
```

Comment:
Nice job here!  I forgot that wp data does not have an easy for us to check dispatch resolution.  This might be something we consider adding to CRUD data stores in a future cycle.